File: demo/quickstart.py
"""BANOS Quickstart — Score annotation files with two lines.

Demonstrates the high-level banos.score() API using the example dataset.
Run from the project root:
    python demo/quickstart.py
"""

from pathlib import Path

import banos
import pandas as pd

DATASET = Path(__file__).parent.parent / "data" / "dataset_human_vs_human"


# ── 1. Single recording ──────────────────────────────────────────────────────
pred = pd.read_csv(DATASET / "Recording_1" / "humanAnnotation_2.csv")
gt = pd.read_csv(DATASET / "Recording_1" / "humanAnnotation_1.csv")

BEHAVIORS = ["attack", "investigation", "mount"]
pred = pred[BEHAVIORS]
gt = gt[BEHAVIORS]

metrics = banos.score(pred, gt)

print("Per-behavior metrics - Recording 1")
print("-" * 48)
for behavior, m in metrics.items():
    f1 = m["f1_score"]
    so = m["so"]
    ic = m["ic"]
    f1_s = f"{f1:.3f}" if f1 is not None else "NaN"
    so_s = f"{so:.3f}" if so is not None else "NaN"
    ic_s = f"{ic:.3f}" if ic is not None else "NaN"
    print(f"  {behavior:<12}  F1={f1_s}  SO={so_s}  IC={ic_s}")

print()

# ── 2. Multiple recordings ────────────────────────────────────────────────────
data = {}
for n in range(1, 11):
    rec_dir = DATASET / f"Recording_{n}"
    pred_file = rec_dir / "humanAnnotation_2.csv"
    gt_file = rec_dir / "humanAnnotation_1.csv"
    if pred_file.exists() and gt_file.exists():
        p = pd.read_csv(pred_file)[BEHAVIORS]
        g = pd.read_csv(gt_file)[BEHAVIORS]
        data[f"Recording_{n}"] = (p, g)

group_metrics, overall_metrics = banos.score(data)

print("Overall metrics - 10 recordings (nanmean across behaviors and recordings)")
print("-" * 48)
for metric, value in overall_metrics.items():
    v_s = f"{value:.4f}" if value is not None else "NaN"
    print(f"  {metric:<12}  {v_s}")

File: demo/quickstart.m
%% BANOS Quickstart -- Score annotation files with one call.
%
% Demonstrates the high-level BANOS_score() API using the example dataset.
%
% Run from the project root:
%   matlab -batch "run('demo/quickstart.m')"
% Or open in MATLAB and click Run.

% Add BANOS to path
addpath(fullfile(fileparts(mfilename('fullpath')), '..', 'matlab', 'BANOS'));

DATASET = fullfile(fileparts(mfilename('fullpath')), '..', 'data', 'dataset_human_vs_human');


%% 1. Single recording ---------------------------------------------------

predTbl = readtable(fullfile(DATASET, 'Recording_1', 'humanAnnotation_2.csv'));
gtTbl   = readtable(fullfile(DATASET, 'Recording_1', 'humanAnnotation_1.csv'));

behaviors = {'attack', 'investigation', 'mount'};
predTbl = predTbl(:, behaviors);
gtTbl   = gtTbl(:, behaviors);

% BANOS Matlab format: row 1 = column headers, rows 2+ = binary data
pred = [predTbl.Properties.VariableNames; num2cell(table2array(predTbl))];
gt   = [gtTbl.Properties.VariableNames;   num2cell(table2array(gtTbl))];

metrics = BANOS_score(pred, gt);

fprintf('Per-behavior metrics -- Recording 1\n');
fprintf('%s\n', repmat('-', 1, 48));
behaviors = fieldnames(metrics);
for i = 1:numel(behaviors)
    beh = behaviors{i};
    m   = metrics.(beh);
    f1  = m.f1_score; so = m.so; ic = m.ic;
    fprintf('  %-12s  F1=%.3f  SO=%.3f  IC=%.3f\n', beh, f1, so, ic);
end
fprintf('\n');


%% 2. Multiple recordings ------------------------------------------------

data = struct();
for n = 1:10
    recDir   = fullfile(DATASET, sprintf('Recording_%d', n));
    predFile = fullfile(recDir, 'humanAnnotation_2.csv');
    gtFile   = fullfile(recDir, 'humanAnnotation_1.csv');
    if exist(predFile, 'file') && exist(gtFile, 'file')
        p = readtable(predFile);
        g = readtable(gtFile);
        p = p(:, behaviors);
        g = g(:, behaviors);
        recName = sprintf('Recording_%d', n);
        data.(recName) = { ...
            [p.Properties.VariableNames; num2cell(table2array(p))], ...
            [g.Properties.VariableNames; num2cell(table2array(g))]  ...
        };
    end
end

[~, overallMetrics] = BANOS_score(data);

fprintf('Overall metrics -- 10 recordings (nanmean across behaviors and recordings)\n');
fprintf('%s\n', repmat('-', 1, 48));
metricNames = fieldnames(overallMetrics);
for i = 1:numel(metricNames)
    name = metricNames{i};
    val  = overallMetrics.(name);
    if isnan(val)
        fprintf('  %-12s  NaN\n', name);
    else
        fprintf('  %-12s  %.4f\n', name, val);
    end
end
